src/test/java/io/wisetime/connector/jira/JiraConnectorPostTimeTest.java

```diff
@@ -312,9 +305,132 @@ void postTime_should_handle_tags_not_matching_project_keys_filter() {
+
+    assertThat(createdWorklogs.get(0).getBody())
+        .as("The diary body should be set to the output of the template formatter")
+        .startsWith(timeGroup.getDescription())
+        .contains("|" + timeRow1.getActivity() + "|" + timeRow1.getDescription() + "|")
+        .contains("|" + timeRow2.getActivity() + "|" + timeRow2.getDescription() + "|")
+        .contains("Total worked time: 41m 6s\n" +
+            "Total chargeable time: 50m\n" +
+            "Experience factor: 50%")
+        .endsWith("The above times have been split across 2 cases and are thus greater than " +
+            "the chargeable time in this case");
+  }
+
+  @Test
+  void postTime_check_narrative_duration_whole_duration_each_tag() {
+    final Tag tag1 = fakeEntities.randomTag("/Jira/");
+    final Tag tag2 = fakeEntities.randomTag("/Jira/");
+
+    final TimeRow timeRow1 = fakeEntities.randomTimeRow().activityHour(2018110110).durationSecs(360);
+    final TimeRow timeRow2 = fakeEntities.randomTimeRow().activityHour(2018110109).durationSecs(360);
+
+    final User user = fakeEntities.randomUser().experienceWeightingPercent(80);
+
+    final TimeGroup timeGroup = expectSuccessfulPostingTime(
+        user, ImmutableList.of(timeRow1, timeRow2), ImmutableList.of(tag1, tag2)
+    )
+        .totalDurationSecs(3000)
+        .durationSplitStrategy(TimeGroup.DurationSplitStrategyEnum.WHOLE_DURATION_TO_EACH_TAG)
+        .narrativeType(TimeGroup.NarrativeTypeEnum.AND_TIME_ROW_ACTIVITY_DESCRIPTIONS);
+
+    assertThat(connector.postTime(fakeRequest(), timeGroup))
+        .as("Valid time group should be posted successfully")
+        .isEqualTo(PostResult.SUCCESS);
+
+    // Verify worklog creation
+    ArgumentCaptor<Worklog> worklogCaptor = ArgumentCaptor.forClass(Worklog.class);
+    verify(jiraDaoMock, times(2)).createWorklog(worklogCaptor.capture());
+    List<Worklog> createdWorklogs = worklogCaptor.getAllValues();
+
+    assertThat(createdWorklogs.get(0).getBody())
+        .as("The diary body should be set to the output of the template formatter")
+        .startsWith(timeGroup.getDescription())
+        .contains("|" + timeRow1.getActivity() + "|" + timeRow1.getDescription() + "|")
+        .contains("|" + timeRow2.getActivity() + "|" + timeRow2.getDescription() + "|")
+        .endsWith("Total worked time: 12m\n" +
+            "Total chargeable time: 50m\n" +
+            "Experience factor: 80%");
+  }
+
+  @Test
+  void postTime_check_narrative_duration_narrative_only() {
+    final Tag tag1 = fakeEntities.randomTag("/Jira/");
+    final Tag tag2 = fakeEntities.randomTag("/Jira/");
+
+    final TimeRow timeRow1 = fakeEntities.randomTimeRow().activityHour(2018110110).durationSecs(420);
+    final TimeRow timeRow2 = fakeEntities.randomTimeRow().activityHour(2018110109).durationSecs(300);
+
+    final User user = fakeEntities.randomUser().experienceWeightingPercent(80);
+
+    final TimeGroup timeGroup = expectSuccessfulPostingTime(
```

Comment:
If you can apply the same changes requested in https://github.com/wisetime-io/wisetime-inprotech-connector/pull/18 it will make more sense here as well.

Reply:
Done. Thanks